server/app/src/androidTest/java/androidx/test/uiautomator/UiScrollableCustom.java

```diff
@@ -24,6 +29,29 @@ public boolean scrollIntoView(UiSelector selector, boolean forceScroll) throws U
                 found = true;
             } else {
                 boolean scrolled = this.scrollForward(100);
+                getDevice().waitForIdle();
+                if (!forceScroll && !scrolled) break;
+            }
+        }
+        return found;
+    }
+
+    public boolean scrollIntoView(UiSelector selector, String targetBySelectorStrategy, boolean forceScroll) throws UiObjectNotFoundException, NoSuchMethodException, IllegalAccessException, InvocationTargetException, InvocationTargetException {
+        Tracer.trace(new Object[]{selector});
+        UiSelector childSelector = this.getSelector().childSelector(selector);
+        Method strategyMethod = By.class.getMethod(targetBySelectorStrategy, String.class);
+        BySelector byselector = (BySelector) strategyMethod.invoke(By.class, (String.valueOf(childSelector)));
+        boolean found = false;
+        if (this.exists(childSelector)) {
+            found = true;
+        }else {
+            for (int x = 0; x < getMaxSearchSwipes(); ++x) {
+                boolean scrolled = this.scrollForward(50);
+                getDevice().waitForIdle();
+                getDevice().wait(Until.findObject(byselector), TIMEOUT);
+                if (this.exists(childSelector)) {
+                    found = true; break;
+                };
```

Comment:
Here you repeat condition from the line 45. I suggest to rewrite this method to avoid repetition. You can do it by doing `wait` in the end of the iteration.